proposals/20220926-contrib-component-guidelines.md

```diff
@@ -0,0 +1,55 @@
+# Guidelines for /contrib Components
+
+**Authors**: Kimonas Sotirchos kimwnasptd@arrikto.com
+
+The motivation behind this proposal is to fully document expectations and
+requirements that componetns under `/contrib` should satisfy. This will make it
+more clear how to use a component, how it integrates with Kubeflow, the problems
+it tries to solve as well as dependency versions.
+
+## Goals
+
+* Document requirements that components under `/contrib` should satisfy
+* Introduce a process for deprecating unmaintained components
+
+## Non-Goals
+
+* Get into the discussion of which components are considered "Kubeflow" components
+    * The assumption until now is that components under the Kubeflow GitHub Org live
+      under the `/apps` dir, and all others under `/contrib`
+* Provide a migration plan for components to move out from `/contrib`
+* Modify the [`example/kustomization.yaml`](https://github.com/kubeflow/manifests/blob/master/example/kustomization.yaml) with new components
+
+## Proposal
+
+### Component Requirements
+
+Components living under `/contrib` should satisfy some strict requirements to
+ensure they are always usable by end-users and contain complementary documentation.
+
+These are the following items that all components under `/contrib` should have:
+1. A `README.md` file that documents:
+    * The K8s version and KF component versions that the component needs
+    * Instructions on how someone can install the component in a Kubeflow cluster
+        * Since Kubeflow manifests have standardized on [Kustomize](https://kustomize.io/)
+          we expect all manifests to be a kustomize packages
+    * How to use the component as part of Kubeflow (examples)
+    * The problems it tries to solve and the value it brings
```

Comment:
Should we also require a link to official documentation/repo?

Reply:
Done.